docker-plain/files/chart/templates/deployment.yaml

```diff
@@ -0,0 +1,70 @@
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: {{ include "chart.fullname" . }}
+  labels:
+    {{- include "chart.labels" . | nindent 4 }}
+spec:
+  {{- if not .Values.autoscaling.enabled }}
+  replicas: {{ .Values.replicaCount }}
+  {{- end }}
+  strategy:
+    type: RollingUpdate
+    rollingUpdate:
+      maxUnavailable: 0%
```

Comment:
is the values of maxUnavailable and maxSurge some best practices we do at BI or how come these are "hardcoded" here? Or are these defaults?

Reply:
As per [docs](https://kubernetes.io/docs/concepts/workloads/controllers/deployment/#rolling-update-deployment), the defaults for both rolling update config params is 25%.
I would make it configurable too, I will provide a new commit for this! good finding @renedupont! 💯

Reply:
@renedupont  @segfault16  please take a look at the new commit https://github.com/opendevstack/ods-quickstarters/pull/1035/commits/0c59afcd5a180ce30200d6856c65a65be8320591 that faces this comment and give your feedback, thanks!

Reply:
Do we specifically need to put both values in the values.yaml if we set them to the default values anyway? Are they that important or often changed to put them there? I am just thinking the more stuff we put in there, the more complex it looks like (especially for kubernetes/helm beginners).

Reply:
Makes sense, I made the proposal to move that there as we want users be enabled on configuration capabilities via values.yaml. So, to having as a goal to (almost) only use values.yaml and require minimal touch (if none at all) on the resources yamls.
I would either then do this way already proposed, or make it more elaborated logic in the deployment yaml like: if you want recreate then it just writes down the recreate statement and if you want rollingupdate then it writes down the rollingupdate statements and the configured/defined required values.
what would you do?
Unless you both prefer to just keep what it was done already and see how this is used later on and, if required, improve from feedback once released. And maybe only do a tiny improvement by pointing to docs and why we propose such values.

Reply:
As just discussed: Lets keep the fields, but don't use the defaults but the proposed values (50% and 0%)